src/Compilers/CSharp/Portable/CodeGen/EmitStackAllocInitializer.cs

```diff
@@ -33,6 +33,11 @@ private void EmitStackAllocInitializers(TypeSymbol type, BoundArrayInitializatio
                 if (data.All(datum => datum == data[0]))
                 {
                     _builder.EmitStackAllocBlockInitializer(data, inits.Syntax, emitInitBlock: true, _diagnostics);
+
+                    if (initializationStyle == ArrayInitializerStyle.Mixed)
```

Comment:
Note: This won't be hit with something like `stackalloc byte[] {1, 1, 1, a, b, c }` while it should. would it worth it?

Reply:
Seems like there are two cases here: 

1. The initializer is completely constant
1. The initializer has non constant values

The only code that truly knows which case is which is `GetRawData`. Could we simplify this to having that return whether or not this was constant and if not do the manual writing of the values?

Reply:
>Could we simplify this to having that return whether or not this was constant

`GetRawData` returns `0` for non constants and the constant zero (which is the source of this bug)

`GetRawData` only helps to know if we can use initblk (same bytes), otherwise we already know about constants via `ShouldEmitBlockInitializerForStackAlloc`. 

So, in order to do a mixed style init with initblk we should examine the resulting byte array: all the none-zero bytes must be the same. And due to the double-entendre of "zero", may not be straightforward.

Reply:
You can go as far as finding all contiguous constant spans longer than, say 5, and memset each of them via `initblk`. I wonder if it is worth the extra effort though. 
Even when we do `cpblk` the codegen is pretty efficient. It is nice to use `initblk` where convenient because "why not", but since this all is allocated on the stack, the arrays are not going to be large and savings from `initblk` are not that huge.

Reply:
>the arrays are not going to be large and savings from initblk are not that huge.

I agree. though I think we can/should do a better job here once SkipLocalsInit is available.

Reply:
Yes, once you can check `AreLocalsZeroed` on the containing method, zero-initing `initblk` may be omitted.